File: src/MaximumAverageSubArrayI.java
public class MaximumAverageSubArrayI {
    public double findMaxAverage(int[] numbers, int k) {
        long sum = 0;
        for (int i = 0 ; i < k ; i++) {
            sum += numbers[i];
        }
        double maxSum = sum;
        for (int index = 0 ; index < numbers.length - k ; index++) {
            sum += numbers[index + k] - numbers[index];
            maxSum = Math.max(maxSum, sum);
        }
        return maxSum / k;
    }
}

File: python/maximum_average_subarray_I.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        moving_sum = sum(nums[i] for i in range(k))
        result = moving_sum
        for i in range(len(nums) - k):
            moving_sum += nums[k + i] - nums[i]
            result = max(result, moving_sum)
        return result / k
